Sources/Prometheus/MetricTypes/Histogram.swift

```diff
@@ -222,21 +230,16 @@ public class PromHistogram<NumType: DoubleRepresentable, Labels: HistogramLabels
         }
         return try body()
     }
-}
 
-extension PrometheusClient {
     /// Helper for histograms & labels
-    fileprivate func getOrCreateHistogram<T: Numeric, U: HistogramLabels>(with labels: U, for histogram: PromHistogram<T, U>) -> PromHistogram<T, U> {
-        let histograms = histogram.subHistograms.filter { (metric) -> Bool in
-            guard metric.name == histogram.name, metric.help == histogram.help, metric.labels == labels else { return false }
-            return true
-        }
-        if histograms.count > 2 { fatalError("Somehow got 2 histograms with the same data type") }
-        if let histogram = histograms.first {
+    fileprivate func getOrCreateHistogram(with labels: Labels) -> PromHistogram<NumType, Labels> {
+        let histogram = self.subHistograms[labels]
+        if let histogram = histogram, histogram.name == self.name, histogram.help == self.help {
             return histogram
         } else {
-            let newHistogram = PromHistogram<T, U>(histogram.name, histogram.help, labels, Buckets(histogram.upperBounds), self)
-            histogram.subHistograms.append(newHistogram)
+            guard let prometheus = prometheus else { fatalError("Lingering Histogram") }
```

Comment:
Hmm can this actually ever happen or just annoyance because that weak there that shoulnd't really manifest in reality?

Reply:
Yes, this is a weak reference thing. I think reference to Prometheus is redundant, I'll try to track down its origins and potentially remove if indeed unnecessary.

Reply:
Ah i see the error just moved around, it was there before.